### PR TITLE
fix: prevent achievement page tabs from deleting draft comments

### PR DESCRIPTION
**Pull request by Loloseiz**
Resolves #4737 

The problem was that useFormDraft's debounce is cancelled when the form unmounts. Also, Inertia navigation doesn't reload drafts because React Hook Form caches defaultValues on initial mount.

Forced a synchronous save to sessionStorage during unmount cleanup, and explicitly reset the form whenever the draftKey changes. As mentioned in the Issue's discussion, it also solves the problems in the "View all comments" page. (https://retroachievements.org/achievement/243928/comments)

### CODE REVIEW

File: resources/js/common/components/CommentList/useSubmitCommentForm.ts [hunk 1]
```diff
@@ -35,11 +38,22 @@ export function useSubmitCommentForm({
...
-    defaultValues: { body: '' },
+    defaultValues: { body: draft.body ?? '' },
   });
 
+  useEffect(() => {
+    const currentDraft = loadDraft<FormValues>(draftKey);
+    form.reset({ body: currentDraft.body ?? '' });
+  }, [draftKey, form]);
+
+  const { clearDraft } = useFormDraft(draftKey, form);
```
Comment (wescopeland) [inline resources/js/common/components/CommentList/useSubmitCommentForm.ts]: I've confirmed this works, however, the implementation suggests `useFormDraft` is broken.

If RHF `defaultValues` caching requires a reset here, then any `useFormDraft` caller with a changing draft key has / will have the same bug. Let's move the key-change rehydration into `useFormDraft` so it resets from base defaults + `loadDraft(key)`, and ensure `clearDraft()` still suppresses the final unmount write.

Comment (Loloseiz) [reply resources/js/common/components/CommentList/useSubmitCommentForm.ts]: I'll refactor `useFormDraft` accordingly as you said, all while double-checking other components using this hook since I guess it could create edge cases if the current behavior isn't the expected one.

Comment (Loloseiz) [reply resources/js/common/components/CommentList/useSubmitCommentForm.ts]: I proceeded with the requested changes, and added tests for `useFormDraft.ts`, as well as tweaked other tests that needed to integrate the `sessionStorage` refresh.

Let met know if this needs further changes :+1:

File: resources/js/common/hooks/useFormDraft.test.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        return form;
+      },
+      { initialProps: { key: 'initial-key' as string | null } },
+    );
+
+    // ACT
+    rerender({ key: null });
+
+    // ASSERT
+    expect(true).toBe(true);
```
Comment (wescopeland) [inline resources/js/common/hooks/useFormDraft.test.ts:133]: https://github.com/RetroAchievements/RAWeb/blob/master/docs/AI_POLICY.md

Comment (Loloseiz) [reply resources/js/common/hooks/useFormDraft.test.ts:133]: Oh, I'm so sorry it's my first time contributing to open source projects, I didn't know about AI policies..

A bit late, but I used Gemini to correct parts of my code and to complete the tests to attain max coverage, I overlooked that to push my code faster since I needed to get back to work and that was stupid of me.

I'll still try to push my manual correction tomorrow even if this doesn't make it right, I'm very sorry as I love the project, I sincerely wanted to help out and I feel bad if my attempt only made you waste your time.

Sorry about that again, next time I'll be more careful

Comment (wescopeland) [reply resources/js/common/hooks/useFormDraft.test.ts:133]: No worries, thanks for your understanding and honesty.
